### PR TITLE
Erc20 transformer

### PR DESCRIPTION
**Pull request by elizabethengelman**
Opening an early PR for feedback, which includes a transformer that currently focuses on the `totalSupply` function on an ERC20 token.

### CODE REVIEW

File: examples/constants/constants.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+//Event Signatures
+var TransferSignature = "0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef"
```
Comment (gustin) [inline examples/constants/constants.go]: Future thought, we could easily derive these from a string of the method name and params.

File: examples/erc20_watcher/config.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+type ContractConfig struct {
+	Address    string
+	Abi        string
+	FirstBlock int64
+	LastBlock  int64
+	Name       string
+}
+
+var DaiConfig = ContractConfig{
+	Address:    "0x89d24a6b4ccb1b6faa2625fe562bdd9a23260359",
```
Comment (gustin) [inline examples/erc20_watcher/config.go]: Can we use the DiaContractAddress string here?

File: examples/erc20_watcher/config.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
...type":"uint256"}],"name":"Mint","type":"event"},{"anonymous":false,"inputs":[{"indexed":true,"name":"guy","type":"address"},{"indexed":false,"name":"wad","type":"uint256"}],"name":"Burn","type":"event"},{"anonymous":false,"inputs":[{"indexed":true,"name":"authority","type":"address"}],"name":"LogSetAuthority","type":"event"},{"anonymous":false,"inputs":[{"indexed":true,"name":"owner","type":"address"}],"name":"LogSetOwner","type":"event"},{"anonymous":true,"inputs":[{"indexed":true,"name":"sig","type":"bytes4"},{"indexed":true,"name":"guy","type":"address"},{"indexed":true,"name":"foo","type":"bytes32"},{"indexed":true,"name":"bar","type":"bytes32"},{"indexed":false,"name":"wad","type":"uint256"},{"indexed":false,"name":"fax","type":"bytes"}],"name":"LogNote","type":"event"},{"anonymous":false,"inputs":[{"indexed":true,"name":"src","type":"address"},{"indexed":true,"name":"guy","type":"address"},{"indexed":false,"name":"wad","type":"uint256"}],"name":"Approval","type":"event"},{"anonymous":false,"inputs":[{"indexed":true,"name":"src","type":"address"},{"indexed":true,"name":"dst","type":"address"},{"indexed":false,"name":"wad","type":"uint256"}],"name":"Transfer","type":"event"}]`,
```
Comment (gustin) [inline examples/erc20_watcher/config.go]: Same with DaiAbiString?

File: examples/erc20_watcher/config.go (hunk 2 above)
Comment (elizabethengelman) [reply examples/erc20_watcher/config.go]: Yep, good call!

File: examples/constants/constants.go (hunk 1 above)
Comment (gustin) [reply examples/constants/constants.go]: Good point. For anonymous events I don't think `topics[0]` is logged at all, how do you define a custom signature for them?

File: cmd/erc20.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	"log"
+	"time"
+)
+
+// erc20Cmd represents the erc20 command
+var erc20Cmd = &cobra.Command{
+	Use:   "erc20",
+	Short: "A brief description of your command",
+	Long: `A longer description that spans multiple lines and likely contains examples
+and usage of using your command. For example:
```
[the remark this answers is not in the document]
Comment (elizabethengelman) [reply cmd/erc20.go]: whoops, forgot to push up a commit! good catch!
c1c563b

File: examples/erc20_watcher/every_block/repository_test.go [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+	})
+
+	Describe("MissingBlocks", func() {
+		It("returns the block numbers for which an associated TokenSupply record hasn't been created", func() {
+			createTokenSupplyFor(repository, blockNumber)
+			newBlockNumber := blockNumber + 1
+			createBlock(newBlockNumber, blockRepository)
+			blocks, err := repository.MissingBlocks(blockNumber, newBlockNumber)
+			anotherNewBlockNumber := newBlockNumber + 1
+			createBlock(anotherNewBlockNumber, blockRepository)
```
[the remark this answers is not in the document]
Comment (elizabethengelman) [reply examples/erc20_watcher/every_block/repository_test.go]: Those last two lines were added to make sure that the method only returns missing blocks between the block numbers given as parameters (`anotherBlock` is outside of that range). But perhaps this should be pulled out into a separate test for clarity.

**Comment by elizabethengelman** [comment]
closing this in favor of https://github.com/8thlight/vulcanizedb/pull/26

**Comment by chapsuk** [comment]
Hi, @elizabethengelman!

How i understand  8thlight/vulcanizedb is private fork of vulcanizedb, any plans to merge erc20 transformer changes to this repo?

Thanks

**Comment by elizabethengelman** [comment]
Hi @chapsuk, thanks for reaching out!

Merges will be based on demand. You can join the [vulcanize gitter](https://gitter.im/vulcanizeio/VulcanizeDB?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge) conversation, if you want it merged soon.

**Pull request closed without merging**